collector/pg_stat_walreceiver.go

```diff
@@ -107,13 +107,13 @@ var (
 		trim(both '''' from substring(conninfo from 'host=([^ ]*)')) as upstream_host,
 		slot_name,
 		status,
-		(receive_start_lsn- '0/0') % (2^52)::bigint as receive_start_lsn,
```

Comment:
These are used in fmt.Sprintf below and I think the % is causing the problem. I can't actually get this collector to finish running the query against my local docker.

Reply:
I was able to reproduce this issue, and I found that calling `hasFlushedLSNRows.Close()` before running the second query fixes it. Presumably there is some deadlock waiting for one cursor to be closed before opening another on the same connection.

Reply:
Probably worth switching these to prepared statements.